compiler/rustc_smir/src/rustc_smir/mod.rs

```diff
@@ -54,6 +54,36 @@ impl<'tcx> Context for Tables<'tcx> {
         self.tcx.sess.source_map().span_to_diagnostic_string(self[span])
     }
 
+    fn get_filename(&self, span: &Span) -> Filename {
+        opaque(
+            &self
+                .tcx
+                .sess
+                .source_map()
+                .span_to_filename(self[*span])
+                .display(rustc_span::FileNameDisplayPreference::Short)
+                .to_string(),
+        )
+    }
+
+    fn get_lines(&self, span: &Span) -> Vec<stable_mir::ty::LineInfo> {
```

Comment:
I would suggest that you return something similar to [`span_to_location_info`](https://github.com/rust-lang/rust/blob/130ff8cb6c3d62ed66daf652cbb5323d3f93c4fc/compiler/rustc_span/src/source_map.rs#L451-L462).

It basically returns the line + column for beginning and end of the span.

Reply:
Perfect! Thanks